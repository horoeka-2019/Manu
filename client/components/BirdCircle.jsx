import React from 'react'
import { Link } from 'react-router-dom'

class BirdCircle extends React.Component {
  id = this.props.id
  name = this.props.name
  image = this.props.image
  render () {
    return (
      <React.Fragment>
        <div className='wrapperBirds'>
          <div className='birdContainer'>
            <Link to={`/profile/${this.id}`}>
              <div className="circleImage"><img className={ this.props.found ? "circleImageFound" : "circleImage"} src={ this.image }></img></div>
            </Link>
            <div className='circleTitle'>{ this.name }</div>
          </div>
        </div>
      </React.Fragment>
    )
  }
}

export default BirdCircle


//

// If found state = true show class circleImageFound otherwise circleImage

