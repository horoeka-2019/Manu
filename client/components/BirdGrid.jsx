import React from 'react'
import { Grid } from 'semantic-ui-react'
// import Circle from './Circle'
import BirdCircle from './BirdCircle'

class BirdGrid extends React.Component {
  render () {
    return (
      <React.Fragment>

        <Grid className='gridWrap' columns={2} doubling>
          {this.props.birds.map(bird => <BirdCircle key={bird.name} name={bird.name} image={bird.image} found={bird.found}/>)}
        </Grid>

      </React.Fragment>
    )
  }
}

export default BirdGrid
