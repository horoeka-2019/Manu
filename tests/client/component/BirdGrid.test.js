import React from 'react'
import { mount } from 'enzyme'
import BirdGrid from '../../../client/components/BirdGrid'
import BirdCircle from '../../../client/components/BirdCircle'
import { HashRouter as Router } from 'react-router-dom'

describe('BirdGrid and BirdCircle Component Tests:', () => {
  // Arrange
  const birds = [{ bird_id: 9, name: 'California Quail', info: 'Quail like to eat seeds...', image: 'public/images/californiaquail.jpg/', found: false }, { bird_id: 10, name: 'Thrush', info: 'Thrush like to usually eat from the ground...', image: 'public/images/thrush.jpg/', found: false }]
  const array = birds.length

  it('The BirdGrid component passes the appropriate info to the BirdCircle component', () => {
    // Act
    const wrapper = mount(<Router><BirdGrid birds={birds}/></Router>)
    console.log(wrapper.html())
    // Assert
    expect(wrapper.find(BirdCircle)).toHaveLength(array)
  })
})
