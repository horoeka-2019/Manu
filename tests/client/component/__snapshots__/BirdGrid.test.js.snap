// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirdGrid and BirdCircle Component Tests: render unchanged from previous snapshot 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <BirdGrid
      birds={
        Array [
          Object {
            "bird_id": 9,
            "found": false,
            "image": "public/images/californiaquail.jpg/",
            "info": "Quail like to eat seeds...",
            "name": "California Quail",
          },
          Object {
            "bird_id": 10,
            "found": false,
            "image": "public/images/thrush.jpg/",
            "info": "Thrush like to usually eat from the ground...",
            "name": "Thrush",
          },
        ]
      }
    >
      <div
        className="gridWrapUpper"
      >
        <Grid
          className="gridWrap"
          columns={2}
          doubling={true}
        >
          <div
            className="ui doubling two column grid gridWrap"
          >
            <BirdCircle
              bird_id={9}
              found={false}
              image="public/images/californiaquail.jpg/"
              info="Quail like to eat seeds..."
              key="California Quail"
              name="California Quail"
            >
              <div
                className="wrapperBirds"
              >
                <div
                  className="birdContainer"
                >
                  <Link
                    to="/profile/9"
                  >
                    <LinkAnchor
                      href="#/profile/9"
                      navigate={[Function]}
                    >
                      <a
                        href="#/profile/9"
                        onClick={[Function]}
                      >
                        <div
                          className="circleImage"
                        >
                          <img
                            className="circleImage"
                            src="public/images/californiaquail.jpg/"
                          />
                        </div>
                      </a>
                    </LinkAnchor>
                  </Link>
                  <div
                    className="circleTitle"
                  >
                    California Quail
                  </div>
                </div>
              </div>
            </BirdCircle>
            <BirdCircle
              bird_id={10}
              found={false}
              image="public/images/thrush.jpg/"
              info="Thrush like to usually eat from the ground..."
              key="Thrush"
              name="Thrush"
            >
              <div
                className="wrapperBirds"
              >
                <div
                  className="birdContainer"
                >
                  <Link
                    to="/profile/10"
                  >
                    <LinkAnchor
                      href="#/profile/10"
                      navigate={[Function]}
                    >
                      <a
                        href="#/profile/10"
                        onClick={[Function]}
                      >
                        <div
                          className="circleImage"
                        >
                          <img
                            className="circleImage"
                            src="public/images/thrush.jpg/"
                          />
                        </div>
                      </a>
                    </LinkAnchor>
                  </Link>
                  <div
                    className="circleTitle"
                  >
                    Thrush
                  </div>
                </div>
              </div>
            </BirdCircle>
          </div>
        </Grid>
      </div>
    </BirdGrid>
  </Router>
</HashRouter>
`;
